examples/ai-ml/ray/main.tf

```diff
@@ -0,0 +1,365 @@
+      "ssm:DescribeParameters"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "irsa_policy" {
+  description = "IAM Policy for IRSA"
+  name_prefix = substr("${module.eks_blueprints.eks_cluster_id}-${local.namespace}-access", 0, 127)
+  policy      = data.aws_iam_policy_document.irsa_policy.json
+}
+
+module "cluster_irsa" {
+  source                     = "../../../modules/irsa"
+  kubernetes_namespace       = local.namespace
+  kubernetes_service_account = "${local.namespace}-sa"
+  irsa_iam_policies          = [aws_iam_policy.irsa_policy.arn]
+  eks_cluster_id             = module.eks_blueprints.eks_cluster_id
+  eks_oidc_provider_arn      = module.eks_blueprints.eks_oidc_provider_arn
+
+  depends_on = [module.s3_bucket]
+}
+
+
+data "kubectl_path_documents" "docs" {
+  pattern = "${path.module}/ray-clusters/*.yaml"
+  vars = {
+    namespace       = local.namespace
+    hostname        = var.eks_cluster_domain == null ? "" : var.eks_cluster_domain
+    account_id      = data.aws_caller_identity.current.account_id
+    region          = local.region
+    service_account = "${local.namespace}-sa"
+  }
+}
+
+# Ugly Terraform issue:
+# See https://github.com/gavinbunney/terraform-provider-kubectl/issues/58#issuecomment-718174177
+data "kubectl_path_documents" "count_hack" {
+  pattern = "${path.module}/ray-clusters/*.yaml"
+  vars = {
+    namespace       = ""
+    hostname        = ""
+    account_id      = ""
+    region          = ""
+    service_account = ""
+  }
+}
+
+resource "kubectl_manifest" "cluster_provisioner" {
+  count     = length(data.kubectl_path_documents.count_hack.documents)
+  yaml_body = element(data.kubectl_path_documents.docs.documents, count.index)
+
+  depends_on = [
+    module.cluster_irsa,
+    module.eks_blueprints_kubernetes_addons
+  ]
+}
+
+
+#---------------------------------------------------------------
+# Monitoring
+#---------------------------------------------------------------
+data "kubectl_path_documents" "prometheus" {
+  pattern = "${path.module}/monitoring/monitor.yaml"
+  vars = {
+    namespace = local.namespace
+  }
+}
+
+resource "kubectl_manifest" "prometheus" {
+  yaml_body = data.kubectl_path_documents.prometheus.documents[0]
+
+  depends_on = [
+    module.eks_blueprints_kubernetes_addons,
+    kubectl_manifest.cluster_provisioner
+  ]
+}
+
+resource "random_password" "grafana" {
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "grafana" {
+  name                    = "grafana"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "grafana" {
+  secret_id     = aws_secretsmanager_secret.grafana.id
+  secret_string = random_password.grafana.result
+}
+
+data "aws_secretsmanager_secret_version" "admin_password_version" {
```

Comment:
since the resource is created here, you shouldn't need the data source and you can just replace all instances of `data.aws_secretsmanager_secret_version.admin_password_version.*` with `aws_secretsmanager_secret_version.admin_password_version.*`